frappe/model/document.py

```diff
@@ -1238,7 +1238,11 @@ def save_version(self):
 
 		version = frappe.new_doc("Version")
 
-		if is_useful_diff := version.update_version_info(self._doc_before_save, self):
+		doc_to_compare = self._doc_before_save
+		if not doc_to_compare and (amended_from := self.get("amended_from")):
+			doc_to_compare = frappe.get_doc(self.doctype, amended_from)
+
+		if is_useful_diff := version.update_version_info(doc_to_compare, self):
```

Comment:
One problems with this:

All rows in all tables are considered "changed" even if they haven't changed at all. 



We might have to add one more handler for amended docs: https://github.com/resilient-tech/frappe/blob/6351f44d0814aa4e43d2749b9d1d8ca348569a39/frappe/core/doctype/version/version.py#L14-L18 

For keeping track of original rows maybe temporary `__old_row_name` or something like that might have to be added for diffing logic.

Reply:
I was being naive lol 😅